### ISSUE TITLE
Add xonsh shell in dynamic

### ISSUE THREAD

**Issue by GouravSardana** [IndianOpenSourceFoundation/dynamic-cli#125]
We have to make autocomplete feature for dynamic. To make it work we need to create a xonsh shell. Still confused ? 
Ref to this - [xonsh](https://xon.sh/)

How does it actually works? 

When a user types dynamic. It will redirect to a different shell on the same terminal and we can use autocomplete in that shell because our normal terminal doesn't allow you to autocomplete.

Ref **first 1 sec** of GIF [When he type gitsome, Another shell got popup [still confused? Tried installing gitsome in local and type gitsome in terminal]]- 

![endgoal](https://user-images.githubusercontent.com/31731827/135447452-cccb50bc-bdf6-4651-9035-d91714abc394.gif)

Credit for xonsh - [xonsh](https://xon.sh/)
Credit for Gitsome - [Gitsome](https://github.com/donnemartin/gitsome)

**Comment by sudojarvis** [comment]
Hi, I would like to work on this issue. Could you please give some head start?
It would be helpful.

**Comment by GouravSardana** [comment]
Go ahead buddy @sudojarvis

**Comment by sudojarvis** [comment]
@ GouravSardana gitsome uses xonsh shell right. I have to do a similar thing right?

**Comment by sudojarvis** [comment]
Do we have to do it from scratch or we can merge it from xonsh.

**Comment by GouravSardana** [comment]
@sudojarvis That’s correct gitsome uses the xonsh shell. 
no no, you don’t have to create it from scratch just leverage the xonsh and try to replicate the flow from gitsome

**Comment by sudojarvis** [comment]
@GouravSardana  could you please assign the issue to me under GSSOC'22.

**Comment by GouravSardana** [comment]
Np. Work on this issue. Don’t worry about assigning the issues. I’ll take care of that

**Comment by sudojarvis** [comment]
https://user-images.githubusercontent.com/75030755/156931165-4c714e0d-e22e-4a6f-a3ff-dde702f1b0c7.mp4

Till now it is auto-completing, what I typed previously. But I have used code from gitsome. Is this fine?

**Comment by GouravSardana** [comment]
@sudojarvis Is this using XONSH ? I couldn't see this going to the xonsh terminal just like in gitsome.
-  If I type gitsome, It'll open up a new customize terminal where we can do autocomplete
- Is this autocomplete based on our history or the argument which we have ?

**Comment by sudojarvis** [comment]
> @sudojarvis Is this using XONSH ? I couldn't see this going to the xonsh terminal just like in git
> * If I type gitsome, It'll open up a new customize terminal where we can do autocomplete
> * Is this autocomplete based on our history or the argument which we have ?

yes, to all your question. And updated the video watch from 2:35.

**Comment by sudojarvis** [comment]
> * Is this autocomplete based on our history or the argument which we have ?
based on our history

**Comment by GouravSardana** [comment]
Okay but i think this autocomplete should work based on both things. One is history and other is args.
As of now, can you make a PR, I’ll review it locally and will give you suggestions accordingly

**Comment by GouravSardana** [comment]
@sudojarvis Any update ?

**Comment by sachin-duhan26** [comment]
Are we still in consensus of adding [xonsh](https://xon.sh/) to the project?
